src/Symfony/Component/ImportMaps/ImportMapManager.php

```diff
@@ -0,0 +1,276 @@
+        $this->createImportMap($env, $provider, true, [], []);
+    }
+
+    /**
+     * @param array<string, PackageOptions> $require
+     * @param string[] $remove
+     */
+    private function createImportMap(Env $env, ?Provider $provider, bool $update, array $require, array $remove): void
+    {
+        $this->loadImportMap();
+        $this->removeFromImportMap($remove);
+
+        $install = [];
+        $packages = [];
+        foreach ($this->importMap ?? [] as $name => $data) {
+            if (isset($data['path'])) {
+                $this->filesystem->mkdir($this->publicAssetsDir);
+                $this->filesystem->copy($this->assetsDir.$data['path'], $this->publicAssetsDir.$this->digestName($name, $data['path']));
+
+                continue;
+            }
+
+            if (!$data['url']) {
+                continue;
+            }
+
+            $packages[$name] = new PackageOptions((bool) ($data['download'] ?? false), $data['preload'] ?? false);
+
+            if (preg_match(self::PACKAGE_PATTERN, $data['url'], $matches)) {
+                $constraint = ($matches['registry'] ?? null) ? "{$matches['registry']}:{$matches['package']}" : $matches['package'];
+
+                if (!$update && ($matches['version'] ?? null)) {
+                    $constraint .= "@{$matches['version']}";
+                }
+
+                $install[$matches['package']] = $constraint;
+            }
+        }
+
+        foreach ($require as $packageName => $packageOptions) {
+            if (preg_match(self::PACKAGE_PATTERN, $packageName, $matches)) {
+                $install[$matches['package']] = $packageName;
+                $packages[$matches['package']] = $packageOptions;
+            }
+        }
+
+        $this->jspmGenerate($env, $provider, $install, $packages);
+
+        $this->filesystem->dumpFile(
+            $this->path,
+            sprintf("<?php\n\nreturn %s;\n", VarExporter::export($this->importMap))
+        );
+    }
+
+    /**
+     * @param string[] $remove
+     */
+    private function removeFromImportMap(array $remove): void {
+        foreach ($remove as $packageName) {
+            if (!isset($this->importMap[$packageName])) {
+                continue;
+            }
+
+            $this->removeIfExists($this->assetsDir.$this->importMap[$packageName]['digest']);
+            unset($this->importMap[$packageName]);
+        }
+    }
+
+    private function jspmGenerate(Env $env, ?Provider $provider, array $install, array $packages): void
+    {
+        if (!$install) {
+            return;
+        }
+
+        $json = [
+            'install' => array_values($install),
+            'flattenScope' => true,
+        ];
+        $provider = $provider?->value ?? $this->provider->value;
+        if ($provider !== Provider::Jspm) {
+            $json['env'] = $provider;
+        }
+
+        $json['env'] = ['browser', 'module', $env->value];
+
+        $response = $this->apiHttpClient->request('POST', '/generate', [
```

Comment:
How does this API behave if env or provider have a not-supported value? I would expect a proper error that'd allow debugging the issue, isn't it? If yes, then that's another reason why we don't need the enums: the API should decide what's valid or not - not us. This would be future proof.